src/internal/packager2/load.go

```diff
@@ -53,6 +54,25 @@ func LoadPackage(ctx context.Context, opt LoadOptions) (*layout.PackageLayout, e
 	isPartial := false
 	switch srcType {
 	case "oci":
+		// this is a special case during inspect. do not pull the full package as it may be very large
+		// default to pulling the sbom for simplicity
+		if opt.Inspect {
+			path, err := pullOCIMetadata(ctx, opt.Source, tmpDir, opt.Shasum, architecture)
+			if err != nil {
+				return nil, err
+			}
+			layoutOpt := layout.PackageLayoutOptions{
+				PublicKeyPath:           opt.PublicKeyPath,
+				SkipSignatureValidation: opt.SkipSignatureValidation,
+				IsPartial:               isPartial,
+				Inspect:                 true,
+			}
+			pkgLayout, err := layout.LoadFromDir(ctx, path, layoutOpt)
+			if err != nil {
+				return nil, err
+			}
+			return pkgLayout, nil
+		}
```

Comment:
Opinion based, but I think it would be better to keep the inspect logic in a separate function. The package you get back from inspect is going to be quite different than any other package you get back (no components) so it feels a bit out of the scope of the function to pick it up. 

Keeping it in it's own function would probably also make it simpler to pull or not pull the SBOM